### ISSUE TITLE
Strange/inconsistent error messages when accessing missing members of relatively-imported modules within class functions

### ISSUE THREAD

**Issue by kwshi** [microsoft/pyright#5907]
**Describe the bug**

Pyright correctly raises an error when I try to access a non-existent member of an imported module: for example, `import b` followed by `b.bar` when `b` has no member named `bar` results in the error
> "bar" is not a known member of module "b"

This is correct so far.

What's weird/unexpected is that, if both of the following hold:
- `b` is relatively imported (e.g., `from . import b`)
- the non-existent access occurs within a function inside a class
then an error is still raised, but the diagnostic message seems to cite the wrong module name (it uses the current module name, instead of the actual module we're trying to access).

**Code or Screenshots**

```python
# b.py (empty)
```

```python
# a.py (in same folder as `b.py`)
from . import b

b.foo # correct error: `"foo" is not a known member of module "a.b"`


class X:
    b.bar # correct error: `"bar" is not a known member of module "a.b"`

    def __init__(self):
        b.baz # weird error: `"baz" is not a known member of module "a"` (I think it should at least be consistent with the other ones, e.g., "a.b"?)
```

![image](https://github.com/microsoft/pyright/assets/3887189/62a48bf6-95c0-4a9d-939d-94420e705f16)

**VS Code extension or command-line**

Neovim LSP; also reproduced using pyright CLI.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.327, which I just published. It will also be included in an upcoming release of pylance.
